src/main/java/core/basesyntax/service/activityhandler/BalanceActivityHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.activityhandler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.Fruit;
+import java.util.Optional;
+
+public class BalanceActivityHandler implements ActivityHandler {
+    private final FruitDao fruitDao;
+
+    public BalanceActivityHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public long changeBalance(Fruit fruit) {
+        Optional<Fruit> fruitFromDB = fruitDao.get(fruit);
+        if (fruitFromDB.isPresent()) {
+            long newBalance = fruitFromDB.get().getBalance() + fruit.getBalance();
+            fruitDao.add(fruit.setAmount(newBalance));
+            return newBalance;
```

Comment:
```suggestion
            Fruit fruitToUpdate = fruitFromDB.get();
            long newBalance = fruitToUpdate.getBalance() + fruit.getBalance();
            fruitToUpdate.setBalance(newBalance);
            fruitDao.save(fruitToUpdate);
            return newBalance;
```
same everywhere where needed